yet another awesome list of bioinformatics links 
===============================================
(a mess to be sorted)
---------------------

_Last update pn 2022-07-21_


* https://pythonnumericalmethods.berkeley.edu/notebooks/Index.html - Python Programming And Numerical Methods: 
A Guide For Engineers And Scientists

* https://mlcourse.ai/book/index.html - Open Machine Learning Course

* https://zwdzwd.github.io/InfiniumAnnotation - comprehensive characterization and annotation of Infinium DNA 
Methylation BeadChip probes (EPIC, HM450, HM27).

* https://diytranscriptomics.com - a hybrid course covering best practices for bulk and single cell RNA-seq 
data analysis, with a primary focus on empowering students to be independent in the use of lightweight and 
open-source software and the R/bioconductor environment.

* https://bioicons.com/ - Bioicons is a free library of open source icons for scientific illustrations using 
vector graphics software such as Inkscape or Adobe Illustrator.

* https://biorender.com/ - BioRender is a web app that enables scientists in biotech, biopharma, and academia 
to create and share professional science figures in minutes (instead of hours!) using a scientifically 
accurate image library.

* https://smart.servier.com/ - Medical Art images.

* https://reactome.org/icon-lib - Library of icons.

* https://www.labicons.net/ - drawings of laboratory objects.

* https://youtu.be/NIUTwCoLIVo - Research Proposal Presentation. How to present a research proposal? Step by 
Step Guide.

* https://www.coursera.org/learn/slides - Presentation skills: Designing Presentation Slides

* https://www.biorxiv.org/content/10.1101/2021.05.26.445798v1.full - ...the Telomere-to-Telomere (T2T) 
Consortium has finished the first truly complete 3.055 billion base pair (bp) sequence of a human genome, 
representing the largest improvement to the human reference genome since its initial release. The new 
T2T-CHM13 reference includes gapless assemblies for all 22 autosomes plus Chromosome X, corrects numerous 
errors, and introduces nearly 200 million bp of novel sequence containing 2,226 paralogous gene copies, 115 of 
which are predicted to be protein coding.

* https://bioinformatics.chat - the bioinformatics chat is a podcast about computational biology, 
bioinformatics, and next generation sequencing.

* https://www.phrasebank.manchester.ac.uk - an academic phrasebook, the must-have collection of academic English 
templates for essay and thesis writing.

* https://jobrxiv.org/ - scientific job posting.

* https://www.ibiology.org/educators-resources/online-course/planning-your-scientific-journey/ - planning Your 
Scientific Journey - a short course (just a few hours) that can help students to reshape their expectation from 
Ph.D.

* https://medium.com/age-of-awareness/12-thing-you-should-know-before-you-start-a-phd-9c064a979e8 - 12 Things 
You Should Know Before You Start A PhD

* https://github.com/ryanlayer/giggle - GIGGLE is a genomics search engine that identifies and ranks the 
significance of shared genomic loci between query features and thousands of genome interval files.

* https://www.gnu.org/software/datamash/ - datamash is a command-line program that performs basic numeric, 
textual, and statistical operations (and even pivot tables!) on input textual data files.

* https://www.nature.com/articles/s41586-021-04332-2 - Peta-scale sequencing effort identifies 132,000 RNA 
viruses we didn't know existed.

* https://python-data-science.readthedocs.io/en/latest/index.html - "This documentation summarizes various 
machine learning techniques in Python". Great for the recap.

* https://www.softwareradius.com/best-bioinformatics-software-and-tools/ - best bioinformatics softwares and 
tools.

* https://www.youtube.com/playlist?list=PLBv09BD7ez_6lYVoZ1RzVcOPIT5Lfjo0Y - must watch lectures about 
clustering algorithms.

* https://www.biostars.org/p/9461665/ - Mamba is a drop-in replacement of Conda and uses the same commands and 
configuration options. Mamba is an almost complete rewrite of Conda in C++, allows for parallel downloading of 
data, and has much faster and better dependency resolving compared to Conda. Yet you should still use Conda 
for its activation and deactivation.

* http://lolaweb.databio.org - LOLAweb is a web server and interactive results viewer for enrichment of 
overlap between a user-provided query region set (a bed file) and a database of region sets. It provides an 
interactive result explorer to visualize the highest ranked enrichments from the database.

* https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7891784/ - ten simple rules for getting started with 
command-line bioinformatics.

* https://www.bioinformatics.recipes - are "bioinformatics recipe" is a collection of commands used to perform
a bioinformatics data analysis task.

* https://stats.stackexchange.com/questions/160479/practical-hyperparameter-optimization-random-vs-grid-search -
random search has a probability of 95% of finding a combination of parameters within the 5% optima with only 
60 iterations. Also compared to other methods it doesn't bog down in local optima.

* https://biodatamining.biomedcentral.com/articles/10.1186/s13040-017-0155-3 - "...we present ten quick tips 
to take advantage of machine learning in any computational biology context, by avoiding some common errors 
that we observed hundreds of times in multiple bioinformatics projects."

* https://learnpracticeandshare.com/awesome-bioinformatics-massive-collection-of-resources/ - Awesome Bioinformatics,
Massive Collection of Resources.

* https://github.com/8080labs/ppscore - The PPS is an asymmetric, data-type-agnostic score that can detect linear 
or non-linear relationships between two columns. The score ranges from 0 (no predictive power) to 1 
(perfect predictive power). It can be used as an alternative to the correlation (matrix). 
As a learning algorithm... use a Decision Tree... The score is calculated on the test sets of a K-fold cross-validation...

* https://en.wikipedia.org/wiki/Relief_(feature_selection) - Relief calculates a feature score for each feature which can then be applied 
to rank and select top scoring features for feature selection.

* https://bioinformatics.chat/stir - feature selection, Relief and STIR.

* https://github.com/EpistasisLab/scikit-rebate - this package includes a scikit-learn-compatible Python implementation of ReBATE, a 
suite of Relief-based feature selection algorithms for Machine Learning. 

* https://www.sciencedirect.com/science/article/pii/S1532046418301412 - "MultiSURF yields the most reliable feature selection performance 
across a wide range of problem types."

* https://biodatamining.biomedcentral.com/articles/10.1186/1756-0381-2-5 - "ReliefF is one such promising algorithm, although it has low 
success rate for noisy datasets when the interaction effect is small."

* https://bit.ly/3HMnTVl - how to use Relief

* https://www.researchgate.net/figure/The-advantages-of-smoothing-a-Points-represent-single-CpG-methylation-estimates_fig3_232007884 - 
kernel smoothing: 

* https://scikit-learn.org/stable/modules/density.html - Epanechnikov Quadratic Kernel outperforms other kernels in most cases

* https://jakevdp.github.io/blog/2013/12/01/kernel-density-estimation/ - Kernel Density Estimation in Python

* https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.savgol_filter.html - a Savitzky-Golay filter

* https://youtu.be/nq6iPZVUxZU - a lecture about UMAP for dimension reduction 

* https://towardsdatascience.com/how-exactly-umap-works-13e3040e1668 - how UMAP works and why it is better than tSNE (open the link in 
incognito mode): 

* https://pair-code.github.io/understanding-umap/ - to see how UMAP parameters affect clustering

* https://umap-learn.readthedocs.io - Python library

* https://bit.ly/3fxr2fp - how to find best parameters for UMAP

* https://www.illumina.com/content/dam/illumina-marketing/documents/applications/ngs-library-prep/ForAllYouSeqMethods.pdf - Massive infographic by Illumina on illustrating how many sequencing techniques work. Techniques cover protein-protein interactions, RNA transcription, RNA-protein interactions, RNA low-level detection, RNA modifications, RNA structure, DNA rearrangements and markers, DNA low-level detection, epigenetics, and DNA-protein interactions. References included.

* https://github.com/crazyhottommy/ChIP-seq-analysis - Resources on ChIP-seq data which include papers, methods, links to software, and analysis from Tommy Tang - the developer of pyflow-ChIPseq and ATACseq pipelines.

* https://liorpachter.wordpress.com/seq/ - a list of papers on various sequencing technologies and assays from a Bren Professor of Computational Biology at Caltech.

* https://bit.ly/3CxA2fm - Proportional Variability as a non-parametric alternative to the coefficient of variation.

* https://github.com/genome/bam-readcount - is a utility that runs on a BAM or CRAM file and generates low-level information about sequencing data at specific nucleotide positions. 
